src/main/java/lotto/domain/Ranks.java

```diff
@@ -0,0 +1,42 @@
+package lotto.domain;
+
+import java.util.EnumMap;
+import java.util.List;
+
+public class Ranks {
+    private final EnumMap<Rank, Integer> ranks = new EnumMap<>(Rank.class);
+
+    private final int purchaseAmount;
+
+    public Ranks(List<Rank> ranks, int purchaseAmount) {
+        init();
+        this.purchaseAmount = purchaseAmount;
+        for (Rank rank : ranks) {
+            this.ranks.put(rank, countRank(rank));
+        }
+    }
+
+    private void init(){
+        for(Rank rank : Rank.values()) {
+            this.ranks.put(rank, 0);
+        }
+    }
+
+    private Integer countRank(Rank rank) {
```

Comment:
wrapper type보다 원시타입을 사용하면 어떨까요?
```suggestion
    private int countRank(Rank rank) {
```